esy-lib/Fs.re

```diff
@@ -311,7 +311,15 @@ let copyFile = (~src, ~dst) =>
 let rec copyPathLwt = (~src, ~dst) => {
   let origPathS = Path.show(src);
   let destPathS = Path.show(dst);
-  let%lwt stat = Lwt_unix.lstat(origPathS);
+  let%lwt stat =
+    // We have observed NPM tarballs contain root directories with no x bit set,
+    // because of which lstat fails. The following line set the x bit of the root directory,
+    // so that the directory contents are searchable
+    try%lwt(Lwt_unix.lstat(origPathS)) {
+    | Unix.Unix_error(Unix.EACCES, _, _) =>
+      let%lwt () = Lwt_unix.chmod(Path.show(Fpath.parent(src)), 0o777);
```

Comment:
My tip would be to look on which permission npm and yarn give to this cases to ensure compatibility

Reply:
You mean just flip the `x` bit on the permission only when necessary?

Reply:
@EduardoRFS @ManasJayanth I checked the yarn code related to untaring
https://github.com/yarnpkg/yarn/blob/master/src/fetchers/tarball-fetcher.js#L133-L136
So for directories the permissions are set as `755` and for files it is `644`
I think I will do the same in `copyPathLwt` function

Reply:
Sounds good, Melwyn.

Reply:
@EduardoRFS I'll leave this thread open for you review this.

Reply:
That's solved for me